### PR TITLE
feat(documentation): add form-check stories

### PR DESCRIPTION
**Pull request by oliverschuerch**

### CODE REVIEW

**Comment by swisspost-devs** [comment]
Preview environments are getting posted here, as soon as they are ready!

**Review by gfellerph — CHANGES_REQUESTED** [review]

File: packages/documentation/src/stories/components/checkbox/checkbox.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    <label htmlFor="ExampleCheckbox_Inline_2" className="form-check-label">{ args.label }</label>
+  </div>,
+  <div key="FormCheck_3" className="form-check form-check-inline">
+    <input id="ExampleCheckbox_Inline_3" className="form-check-input" type="checkbox"/>
+    <label htmlFor="ExampleCheckbox_Inline_3" className="form-check-label">{ args.label }</label>
+  </div>,
+  <div key="FormCheck_4" className="form-check form-check-inline">
+    <input id="ExampleCheckbox_Inline_4" className="form-check-input" type="checkbox"/>
+    <label htmlFor="ExampleCheckbox_Inline_4" className="form-check-label">{ args.label }</label>
+  </div>
```
Comment (gfellerph) [inline packages/documentation/src/stories/components/checkbox/checkbox.stories.tsx:265]: Can you wrap this example in a `<fieldset><legend>...</legend> ...checkboxes... </fieldset>` structure to have a label for the group of checkboxes?

This should probably also be mentioned in the docs (https://www.accessibility-developer-guide.com/examples/forms/grouping-with-fieldset-legend/).

File: packages/documentation/src/stories/components/checkbox/checkbox.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        'bg-aubergine-bright',
+        'bg-success',
+        'bg-warning',
+        'bg-danger',
+        'bg-info'
+      ],
+      table: {
+        category: 'General'
+      }
+    },
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/checkbox/checkbox.stories.tsx:148]: But we are already communicating this in the current documentation... 🤔

[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/documentation/src/stories/components/checkbox/checkbox.stories.tsx:133]: I know, it's not ideal. I don't think we can ensure that all components work on all backgrounds so I'd like to concentrate on the dark mode and if we accidentally make something work on other backgrounds that's a bonus.

**Review by gfellerph — CHANGES_REQUESTED** [review]
Not sure if intentional, when toggling between indeterminate, unchecked and checked, the visual state will stay indeterminate. I think the controls should correctly represent the visual state.

**Review by alizedebray — CHANGES_REQUESTED** [review]
It would be nice to:
- [x] Add a control to toggle the visibility of the fieldset legend

Apply some changes requested in the form switch PR:
- [x] Update documentation wording and correct typos
- [x] Remove `<p>` and `<div>` elements from argument descriptions
- [ ] Use `class` instead of `className` in HTML strings
- [ ] Show form feedback controls only when necessary

File: packages/documentation/src/stories/components/checkbox/checkbox.docs.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+@use '@swisspost/design-system-styles/components/form-feedback.scss';
+```
+
+## Specific examples of application
+
+The following examples show the different characteristics of the component. These can differ in the type of visualization, the html structure, as well as when, how and why they are displayed.
+
+### Indeterminate
+
+Normally a checkbox input can only have two states: `checked` or `unchecked`. They can have any value, but they either submit that value (checked) or don't (unchecked) with a form submission.
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/checkbox/checkbox.docs.mdx:45]: I would prefer to leave this text as is, if this is ok for you?
Source: https://bootstrap-vue.org/docs/components/form-checkbox#indeterminate-tri-state-support

File: packages/documentation/src/stories/components/checkbox/checkbox.docs.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+Visually, there are three states a checkbox can be in: `checked`, `unchecked`, or `indeterminate`.<br/>
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/checkbox/checkbox.docs.mdx:47]: I would prefer to leave this text as is, if this is ok for you?
Source: https://bootstrap-vue.org/docs/components/form-checkbox#indeterminate-tri-state-support

File: packages/documentation/src/stories/components/checkbox/checkbox.docs.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+The `indeterminate` state is visual only. The checkbox is still either **checked** or **unchecked** as a value. That means the visual `indeterminate` state masks the real value of the checkbox. A possible use case for this state could be a list of checkboxes with one to toggle them all as in a folder/subfolder tree and only a part of the subfolders are selected. For an example visit [Accessible Web Designs indeterminate checkbox](http://www.barrierefreies-webdesign.de/knowhow/checkbox/indeterminate.html).
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/checkbox/checkbox.docs.mdx:48]: I would prefer to leave this text as is, if this is ok for you?
Source: https://bootstrap-vue.org/docs/components/form-checkbox#indeterminate-tri-state-support

**Review by gfellerph — APPROVED** [review]
Amazing! It's coming together nicely.

**Pull request merged**